cmd/gateway-sia.go

```diff
@@ -0,0 +1,431 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/minio/minio-go/pkg/policy"
+	"io"
+	"os"
+	"strconv"
+)
+
+type siaObjects struct {
+	Cache              *SiaCacheLayer // Cache layer.
+	Fs                 *fsObjects     // Filesystem layer.
+	PurgeCacheAfterSec int64          // How many seconds to wait to purge object from SiaBridge cache.
+	SiadAddress        string         // Address and port of Sia Daemon.
+	CacheDir           string         // Name of cache directory.
+	DbFile             string         // Name of cache database file.
+	DebugMode          bool           // Whether or not debug mode is enabled.
+}
+
+// Convert Sia errors to minio object layer errors.
+func siaToObjectError(err *SiaServiceError, params ...string) (e error) {
+	if err == nil {
+		return nil
+	}
+
+	switch err.Code {
+	case "SiaErrorDaemon":
+		return fmt.Errorf("Sia Daemon: %s", err.Message)
+	case "SiaErrorUnableToClearAnyCachedFiles":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDeterminingCacheSize":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDatabaseDeleteError":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDatabaseCreateError":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDatabaseCantBeOpened":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDatabaseInsertError":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDatabaseUpdateError":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorDatabaseSelectError":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaFailedToDeleteCachedFile":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorObjectDoesNotExistInBucket":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorObjectAlreadyExists":
+		return fmt.Errorf("Sia: %s", err.Message)
+	case "SiaErrorUnknown":
+		return fmt.Errorf("Sia: %s", err.Message)
+	default:
+		return fmt.Errorf("Sia: %s", err.Message)
+	}
+}
+
+// newSiaGateway returns Sia gatewaylayer
+func newSiaGateway(host string) (GatewayLayer, error) {
+	sia := &siaObjects{
+		PurgeCacheAfterSec: 24 * 60 * 60,
+		SiadAddress:        "127.0.0.1:9980",
```

Comment:
Change made as requested so that the Sia daemon api address can be specified on command line.